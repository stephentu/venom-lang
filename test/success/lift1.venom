def f() =
  a = 10;
  b = 10;
  def g() = 
    a = 20; # this refers to the *outer* a
    b::int = 20; # this is a new decl
    print (a);
    print (b);
  end
  g();
  print (a);
  print (b);
end
f();
