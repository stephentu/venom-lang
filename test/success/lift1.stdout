20
20
20
10
